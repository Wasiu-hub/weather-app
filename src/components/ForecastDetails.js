/* eslint-disable prettier/prettier */
/* eslint-disable react/no-unknown-property */
import React from "react";

function ForecastDetails({ forecast }) {
  const { date, humidity, temperature, wind } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-summary">
      <div className="forecast-details">{formattedDate}</div>

      <div className="forecast-details__temperature">
        Max Temperature: {temperature.max} &deg;C
      </div>
      <div className="forecast-details__temperature">
        Min Temperature: {temperature.min} &deg;C
      </div>
      <div className="forecast-details__wind">Wind Speed: {wind.speed}mph</div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
    </div>
  );
}

export default ForecastDetails;
