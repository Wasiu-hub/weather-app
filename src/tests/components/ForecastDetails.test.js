/* eslint-disable prettier/prettier */
import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: "Stub humidity",
    temperature: {
      min: 12,
      max: 22,
    },
    wind: "Stub wind",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
