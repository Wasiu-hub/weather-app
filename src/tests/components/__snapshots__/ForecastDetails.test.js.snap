// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-details"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      Max Temperature: 22 °C
    </div>
    <div
      class="forecast-details__temperature"
    >
      Min Temperature: 12 °C
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind Speed: mph
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: Stub humidity%
    </div>
  </div>
</DocumentFragment>
`;
